x/profile/internal/types/msgs.go

```diff
@@ -0,0 +1,218 @@
+	PreviousMoniker string         `json:"previous_moniker"`
+	NewMoniker      string         `json:"new_moniker"`
+	Name            string         `json:"name,omitempty"`
+	Surname         string         `json:"surname,omitempty"`
+	Bio             string         `json:"bio,omitempty"`
+	ProfilePic      string         `json:"profile_pic,omitempty"`
+	ProfileCov      string         `json:"profile_cov,omitempty"`
+	Creator         sdk.AccAddress `json:"creator"`
+}
+
+// NewMsgEditProfile is a constructor function for MsgEditProfile
+func NewMsgEditProfile(previousMoniker string, newMoniker string, name string, surname string, bio string, profilePic string,
+	profileCov string, creator sdk.AccAddress) MsgEditProfile {
+	return MsgEditProfile{
+		PreviousMoniker: previousMoniker,
+		NewMoniker:      newMoniker,
+		Name:            name,
+		Surname:         surname,
+		Bio:             bio,
+		ProfilePic:      profilePic,
+		ProfileCov:      profileCov,
+		Creator:         creator,
+	}
+}
+
+// Route should return the name of the module
+func (msg MsgEditProfile) Route() string { return RouterKey }
+
+// Type should return the action
+func (msg MsgEditProfile) Type() string { return ActionEditProfile }
+
+// ValidateBasic runs stateless checks on the message
+func (msg MsgEditProfile) ValidateBasic() error {
+	if msg.Creator.Empty() {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, fmt.Sprintf("Invalid creator address: %s", msg.Creator))
+	}
+
+	if len(strings.TrimSpace(msg.PreviousMoniker)) == 0 {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "Profile previous moniker cannot be blank or empty")
+	}
+
+	if len(msg.PreviousMoniker) > MaxMonikerLength {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("Profile previous moniker cannot exceed %d characters", MaxMonikerLength))
+	}
+
+	if len(msg.NewMoniker) > MaxMonikerLength {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("Profile new moniker cannot exceed %d characters", MaxMonikerLength))
+	}
+
+	if len(msg.Name) != 0 && len(msg.Name) < MinNameSurnameLength {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("Profile name cannot be less than %d characters", MinNameSurnameLength))
+	}
+
+	if len(msg.Name) > MaxNameSurnameLength {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("Profile name cannot exceed %d characters", MaxNameSurnameLength))
+	}
+
+	if len(msg.Surname) != 0 && len(msg.Surname) < MinNameSurnameLength {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("Profile surname cannot be less than %d characters", MinNameSurnameLength))
+	}
+
+	if len(msg.Surname) > MaxNameSurnameLength {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("Profile surname cannot exceed %d characters", MaxNameSurnameLength))
+	}
+
+	if len(msg.Bio) > MaxBioLength {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("Profile biography cannot exceed %d characters", MaxBioLength))
+	}
+
```

Comment:
Probably we should also add a check to both profile and cover pictures making sure that, if they exist, they are valid URIs.

Reply:
As I said to you on last friday call,  I will pass the `default` string value instead.
I've done this just to keep the `cli` commands clean otherwise I should have put a lot of if statements just to check if a field has the default and than put it to nil before build the message.

Reply:
@bragaz Please go [here](https://github.com/desmos-labs/desmos/pull/122#discussion_r404590307)